internal/monitors/kubernetes/cluster/metrics/pods.go

```diff
@@ -50,56 +51,68 @@ func datapointsForPod(pod *v1.Pod) []*datapoint.Datapoint {
 	return dps
 }
 
-func dimPropsForPod(cachedPod *k8sutil.CachedPod, sc *k8sutil.ServiceCache,
-	rsc *k8sutil.ReplicaSetCache, jc *k8sutil.JobCache) *atypes.DimProperties {
-	props, tags := k8sutil.PropsAndTagsFromLabels(cachedPod.LabelSet)
+func dimensionForPod(pod *v1.Pod) *atypes.Dimension {
+	props, tags := k8sutil.PropsAndTagsFromLabels(pod.Labels)
 
-	props["pod_creation_timestamp"] = cachedPod.CreationTimestamp.Format(time.RFC3339)
+	props["pod_creation_timestamp"] = pod.CreationTimestamp.Format(time.RFC3339)
 
-	for _, or := range cachedPod.OwnerReferences {
+	for _, or := range pod.OwnerReferences {
 		props["kubernetes_workload"] = or.Kind
 		props["kubernetes_workload_name"] = or.Name
 		props[utils.LowercaseFirstChar(or.Kind)] = or.Name
 		props[utils.LowercaseFirstChar(or.Kind)+"_uid"] = string(or.UID)
 	}
 
-	_ = getPropsFromTolerations(cachedPod.Tolerations)
+	_ = getPropsFromTolerations(pod.Spec.Tolerations)
 
-	// if pod is selected by a service, sync service as a tag
-	serviceTags := sc.GetMatchingServices(cachedPod)
-	for _, tag := range serviceTags {
-		tags["kubernetes_service_"+tag] = true
+	if len(props) == 0 && len(tags) == 0 {
+		return nil
```

Comment:
is this check required? it seems redundant since we set `MergeIntoExisting` to true.

Reply:
No, and in fact returning nil is very bad so I will remove this whole conditional.